packages/sdk/src/modules/unification/index.ts

```diff
@@ -0,0 +1,138 @@
+  message: string;
+  evm: string;
+  substrate: string;
+}
+
+enum ChainID {
+  AGUNG = 9990,
+  KREST = 2241,
+  PEAQ = 3338,
+}
+
+
+export class Unification extends Base {
+    constructor(
+      protected override readonly _api?: ApiPromise,
+      protected readonly _metadata?: SDKMetadata,
+  
+    ) {
+      super();
+    }
+
+  /**
+   * Binds a SS58 Substrate wallet to a newly created H160 Ethereum wallet using the address unification pallet peaq provides.
+   * Make sure the H160 wallet is new and has no transactions.
+   * 
+   * @param options ClaimAccountOptions - The options for binding the addresses.
+   * @returns CreateDidResult - Contains the block_hash of the executed transaction and unsubscribe() to terminate event listening.
+   */
+  public async claimAccount(options: ClaimAccountOptions, statusCallback?: (result: ISubmittableResult) => void | Promise<void>): Promise<ClaimAccountResult> {
+    try {
+        const api = this._getApi();
+
+        const { network, substrateSeed, ethPrivate } = options;
+        if (!network) throw new Error("Error: No network provided.");
+        if (!substrateSeed) throw new Error("Error: No substrate private key provided.");
+        if (!ethPrivate) throw new Error("Error: No ethereum private key provided.");
+
+        const keyPair = this._getKeyPair(substrateSeed);
+        const ss58Address = keyPair.address;
+
+        const signer = new ethers.Wallet(ethPrivate);
+        const evmAddress = signer.address;
+
+        const ethSignature = await this._createKeyBind(network, ss58Address, signer);
+
+        const attributeExtrinsic = api.tx?.['addressUnification']?.['claimAccount'](
+          evmAddress,
+          ethSignature
+        );
+
+        const nonce = await this._getNonce(keyPair.address);
+        const eventData = await this._newSignTx({nonce, address: keyPair, extrinsics: attributeExtrinsic});
+        const unsubscribe = await attributeExtrinsic.send((result) => {
+          statusCallback &&
+            statusCallback(result as unknown as ISubmittableResult);
+        });
+
+        return {
+          message: "Address Unification Successful.",
+          evm:  `${evmAddress}`,
+          substrate: `${ss58Address}`
+      }
+    } catch (error) {
+      if (typeof error === 'object' && error !== null && 'data' in error) throw new ClaimAccountError(`${error.data}`);
+      throw new ClaimAccountError(`${error}`);
+    }
+  }
+
+  protected async _createKeyBind(network: string, ss58Address: string, signer: Wallet): Promise <string>{
+    try {
+      const api = this._getApi();
+
+      // get chain name from api to check against what the user manually set
+      const properties = await api.rpc.system.properties();
+      const readable = properties.toHuman();
+
+      // want to make sure the set network is a known enum, and the set network is the same as the api metadata
+      if (ChainID[network.toUpperCase() as keyof typeof ChainID] == undefined 
```

Comment:
This is where the check takes place. It first makes sure the network string is in the enum, and then compares to the currently connected api in the metadata to see if they align.